lib/codebreaker_gem/game.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+require_relative 'validation/validation'
+require_relative 'validation/errors_include'
+
+module CodebreakerGem
+  class Game
+    include Validation
+    attr_accessor :name, :code, :difficulty, :attempts, :hints, :available_hints
+
+    CODE_RANGE = (1..6).to_a.freeze
+    CODE_LENGTH = 4
+    DIFFICULTIES = {
+      easy: { attempts: 15, hints: 2 },
+      medium: { attempts: 10, hints: 1 },
+      hell: { attempts: 5, hints: 1 }
+    }.freeze
+
+    def initialize
+      @name = ''
+      @code = ''
+      @difficulty = 0
+      @attempts = 0
+      @hints = 0
+    end
+
+    def start
+      @code = CODE_RANGE.sample(CODE_LENGTH).join
```

Comment:
You defined accessor for 'code' attribute, so you can use it for save value without sign @. But you should find information how to it work) Call attributes via @ only if you don`t create any accessor.